syntax = "proto3";

import "google/protobuf/empty.proto";
option java_multiple_files = true;
option java_package = "demo.microservices.sync.grpc";

// ---------------Recommendation service----------

service RecommendationService {
  rpc ListRecommendations(RecommendationsRequest) returns (ListRecommendationsResponse){}
  rpc HealthCheck(google.protobuf.Empty) returns (HealthStatusResponse){};
}

message RecommendationsRequest {
    repeated string product_category = 1;
}

message ListRecommendationsResponse {
    repeated string product_ids = 1;
}

message HealthStatusResponse {
    string status = 1;
}